### PR TITLE
Do not use fragile native image content when generating R2R images

### PR DESCRIPTION
**Pull request by jkotas**
Port https://github.com/dotnet/coreclr/pull/5715 to release/1.0.0 to fix https://github.com/dotnet/coreclr/issues/5720

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT Checked pri1r2r please

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu Checked pri1r2r please

**Comment by jkotas** [comment]
@gkhanna79 PTLA

File: src/vm/clsload.cpp [hunk 1]
```diff
@@ -100,8 +100,9 @@ PTR_Module ClassLoader::ComputeLoaderModuleWorker(
 
 #ifndef DACCESS_COMPILE
 #ifdef FEATURE_NATIVE_IMAGE_GENERATION
-    // Check we're NGEN'ing
-    if (IsCompilationProcess())
+    // Check we're NGEN'ing. ComputeLoaderModuleForCompilation algorithm assumes that there is fragile
+    // NGen image for CoreLib that is not the case for ReadyToRun compilation.
```
Comment (gkhanna79) [inline src/vm/clsload.cpp]: NIT: CoreLib always has fragile Ngen image. Shouldn't the comment say something like "ComputeLoaderModuleForCompilation assumes we can save information from fragile CoreLib into R2R image being generated, which is not version resilient incase we wish to mix and match Check corelib with R2R images generated against release CoreLib. Hence, we skip sving such information for R2R" ?

**Comment by gkhanna79** [comment]
LGTM

**Comment by gkhanna79** [comment]
Failure is known issue (http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/release_1.0.0/job/x64_checked_ubuntu_pri1r2r_tst_prtest/8/testReport/).

File: src/vm/clsload.cpp (hunk 1 above)
Comment (jkotas) [inline src/vm/clsload.cpp]: https://github.com/dotnet/coreclr/pull/5738

**Pull request merged**